### DISCUSSION TITLE
Document does not work

### DISCUSSION THREAD

**Discussion by ThinhNguyen-Madi** [feathersjs/feathers#3007] [Q&A]
I write flow: https://feathersjs.com/guides/basics/starting.html#an-api-server
It display error:
`app.ts:45:25 - error TS2339: Property 'on' does not exist on type 'FeathersService<Application<ServiceTypes, any>, MessageService>'.

45 app.service('messages').on('created', (message: Message) => {`

**Comment by daffl** [comment]
You likely ended up installing the latest version instead of the current prerelease.

```
npm ls @feathersjs/feathers
```

Should show `5.0.0-pre` (and not `4.5.x`). The installation command shown in the guide should give you the correct version (note the `@pre`):

```
npm install @feathersjs/feathers@pre --save
```

**Reply by nandin-borjigin** [reply]
~~5.0.0-pre.36 also shows the same error~~

It seems like I missed `@types/node` and `on` method is inherited from `EventEmitter`. After install `@types/node` the problem goes away.

![image](https://user-images.githubusercontent.com/10903833/216295940-07775910-a76e-4d39-879d-ef1fa7b848bb.png)

**Reply by daffl** [reply]
Ah, thank you for digging into this. I added it the `@types/node` package to the installation instructions in the guide.
